internal/provider/resource_tfe_data_retention_policy.go

```diff
@@ -0,0 +1,368 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-framework-validators/numbervalidator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/objectvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/numberplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"strings"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &resourceTFEDataRetentionPolicy{}
+var _ resource.ResourceWithConfigure = &resourceTFEDataRetentionPolicy{}
+var _ resource.ResourceWithImportState = &resourceTFEDataRetentionPolicy{}
+
+func NewDataRetentionPolicyResource() resource.Resource {
+	return &resourceTFEDataRetentionPolicy{}
+}
+
+// resourceTFEDataRetentionPolicy implements the tfe_data_retention_policy resource type
+type resourceTFEDataRetentionPolicy struct {
+	config ConfiguredClient
+}
+
+func (r *resourceTFEDataRetentionPolicy) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_data_retention_policy"
+}
+
+func (r *resourceTFEDataRetentionPolicy) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Manages the data retention policies for a specific workspace or an the entire organization.",
+		Version:     1,
+
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description: "ID of the Data Retention Policy.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"organization": schema.StringAttribute{
+				Description: "Name of the organization. If omitted, organization must be defined in the provider config.",
+				Optional:    true,
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"workspace_id": schema.StringAttribute{
```

Comment:
I am able to generate errors by creating a config with multiple policies specified for the same workspace.

```
resource "tfe_data_retention_policy" "workspace1_policy" {
  workspace_id = tfe_workspace.workspace1.id
  delete_older_than {
    days = 3
  }
}

resource "tfe_data_retention_policy" "workspace1_policy2" {
  workspace_id = tfe_workspace.workspace1.id
  delete_older_than {
    days = 6
  }
}
```


If I start with one policy and then add the other, the policy is replaced with the new value (e.g. it goes from 3 days to 6 days). Then every subsequent apply detects drift, and tries to update the policy back to the other value (3 -> 6 -> 3 -> 6 -> .....)

If I try to modify both policies at once, I get an error as it attempts to delete the same policy twice.

Reply:
If I delete one of the configs, it deletes the policy on TFE (as expected). Then further applies crash with the error

```
│ The plugin.(*GRPCProvider).ReadResource request was cancelled.
╵

Stack trace from the terraform-provider-tfe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x1027f0434]

```

Reply:
Good catch 👍 I knew having multiple resources "overriding each other" on a given workspace or organization would be problematic, but we've ignored that problem in the past. That being said, I didn't expect the removal to cause a nasty error like that. I pushed a "fix" that at least allows for users to remove policies attached to the same workspace/organization.